### ISSUE TITLE
Freeze during validation with distributed training and model with batch normalization layers

### ISSUE THREAD

**Issue by SweetVlad** [pytorch/pytorch#20611]
## ❓ Questions and Help
Hi,
I got unexpected behavior during training with torch.distributed.DistributedDataParallel model on multiple GPUs.

- I train my model with DistributedSampler and DataLoader from torch lib. During training everything is fine, but when I validate my model after the first epoch of training, everything gets stuck  - no crash, but GPUs are showing 100 % utilization as well as CPUs, but nothing happens.

- If I remove BatchNorm2d layers from the model, everything runs just fine.

- If I use model with BatchNorm2d layers with world size 1 (only one GPU but setup through DistributedDataParallel), everything is also fine.

I assume that different batchnorm values are not correctly synced across gpus or any idea what am I doing wrong?

Thanks.

## version / os
dist backend -"nccl", Pytorch 1.1, Cuda 10, ubuntu 16.04

**Comment by jerryzh168** [comment]
Please use https://discuss.pytorch.org (https://discuss.pytorch.org/) for questions. Please kindly feel encouraged to reopen this issue, if this does not apply.

**Comment by GeoffreyChen777** [comment]
I have a similar problem.

My training will get stuck at the same iteration steps. The GPU usage is 100%, but nothing happens.

**Comment by YYuanAnyVision** [comment]
pytorch 1.1
the same problem here, the training freezes after 20 epochs. GPU usage is constantly 100%( in the normal situation, it would jump between 80% to 100%), the data loader also stops working.

**Comment by XiaosuTong** [comment]
running into this issue as well. Notice that it is not training freezing, it is hanging during validation. I also noticed that if I turned the track_running_stats to be False, the problem went away. But this make the validation risky
